packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -172,14 +175,28 @@ export default abstract class SqlIntegration
 
   private wrapRunQuery() {
     const originalRunQuery = this.runQuery;
-    this.runQuery = async (sql: string, setExternalId?: ExternalIdCallback) => {
+    this.runQuery = async (
+      sql: string,
+      setExternalId?: ExternalIdCallback,
+      metadata?: QueryMetadata,
+    ) => {
       if (isMultiStatementSQL(sql)) {
         throw new Error("Multi-statement queries are not supported");
       }
-      return originalRunQuery.call(this, sql, setExternalId);
+      metadata = {
+        ...metadata,
+        userId: this.context.userId,
+        userName: this.context.userName,
```

Comment:
nit: Should this happen when we create the Integration and set it in `additionalMetadata` property?

I am happy to skip this for now as well as it creates the possibility of overriding it, etc, and we would need to handle that -- but it would be nice to have a single place with the default context metadata (application, user).

Reply:
Maybe, but we create the integration in a ton of places, so I like the idea of it always being set.